### ISSUE TITLE
Specific Playlist crashes the program

### ISSUE THREAD

**Issue by n30phyte** [xou816/spot#125]
**Describe the bug**
Spot crashes when trying to load playlist.

While I'm waiting for #51, I decided to just make a mirror of my library as a playlist to use in Spot, but it's crashing instantly. From looking at the stack trace, it might be due to songs from the recent Korean music purge being in the playlist, or could be due to it's massive size (~1440 Songs).

**To Reproduce**
1. Have a spotify:playlist:7GI7ETa7E6Hr5G1F5CAOSw saved in your account
2. Click on the playlist from the playlists screen
3. Instant crash

**Expected behavior**
Playlist loads and I can listen to my music

**General information:**
 - Distribution: Fedora 33
 - Installation method: Flatpak installed from Flathub
 - Version: 0.1.9
 - Device used: Desktop

**Stack trace:**
```
❯ flatpak run --env=RUST_BACKTRACE=full dev.alextren.Spot
Gtk-Message: 23:57:26.811: Failed to load module "xapp-gtk3-module"
Gtk-Message: 23:57:26.872: Failed to load module "canberra-gtk-module"
Gtk-Message: 23:57:26.873: Failed to load module "canberra-gtk-module"
home
playlist_7GI7ETa7E6Hr5G1F5CAOSw
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/api/api_models.rs:228:63
stack backtrace:
   0:     0x55c94616a410 - <unknown>
   1:     0x55c94618debc - <unknown>
   2:     0x55c946162ff2 - <unknown>
   3:     0x55c94616c6b5 - <unknown>
   4:     0x55c94616c20a - <unknown>
   5:     0x55c94616ce51 - <unknown>
   6:     0x55c94616c967 - <unknown>
   7:     0x55c94616a8cc - <unknown>
   8:     0x55c94616c8f9 - <unknown>
   9:     0x55c94618b6d1 - <unknown>
  10:     0x55c94618b61d - <unknown>
  11:     0x55c9459c4cab - <unknown>
  12:     0x55c9459b3aa6 - <unknown>
  13:     0x55c945971042 - <unknown>
  14:     0x55c945a25c47 - <unknown>
  15:     0x55c945a38aa3 - <unknown>
  16:     0x55c945a3db2c - <unknown>
  17:     0x55c9459ee5ee - <unknown>
  18:     0x55c94613dd85 - <unknown>
  19:     0x55c94613f5ca - <unknown>
  20:     0x7fc92c21b53b - g_main_context_dispatch
  21:     0x7fc92c21b7e8 - <unknown>
  22:     0x7fc92c21b8b3 - g_main_context_iteration
  23:     0x7fc92c44358d - g_application_run
  24:     0x55c94595a94e - <unknown>
  25:     0x55c94598e4e8 - <unknown>
  26:     0x7fc92c21c665 - g_main_context_invoke_full
  27:     0x55c945a02e71 - <unknown>
  28:     0x55c945985503 - <unknown>
  29:     0x55c9459c96a9 - <unknown>
  30:     0x55c94616d277 - <unknown>
  31:     0x55c945a031e2 - main
  32:     0x7fc92bb06062 - __libc_start_main
  33:     0x55c9458d913e - <unknown>
  34:                0x0 - <unknown>
```

**Additional context**
The playlist is public I think, so you can use it to test.
Nothing else crashes while in use, just that specific playlist.
There doesn't seem to be a convenient way to delete the empty songs ([example here](https://open.spotify.com/album/4pR0eatWzDISWS8LHv5dzZ?highlight=spotify:track:0jnUqXkSxqe8hLkPVfzIjz)) from the playlist.

**Comment by xou816** [comment]
Ah, I thought I had fixed that, but I wasn't paying attention. I'll have a look, thanks for the report!

**Comment by xou816** [comment]
Fixed in 0.1.10
